### PR TITLE
Refactor/wagmi v1

### LINKED ISSUE #349: [Feature] Move from EtherJs to viem, by using wagmi
## Description

We need to move from etherjs to Viem, and use wagmi on top of it. 

## Acceptance Criteria

- [x] Wagmi setup and first use.
- [ ] Refactoring all existing feature with Wagmi

### PR DESCRIPTION
**Pull request by dasarathg68**
# Description
This PR demoes the use of Wagmi hooks to useReadContract, useWriteContract, useWaitForTransactionReceipt useBalance. 
We use a combination of useWriteContract and useWaitForTransactionReceipt to provide better UX.
**Right now, all functions in Bank and Proposals have been refactored to Wagmi** 

Fixes #349 

## Contribution

For your PR please add a commnent to eache file edited to explain the changes you made.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist

Make sur you run the following commands before submitting your PR:

- /app

```bash
npm run build
npm run test:unit
npm run type-check
npm run lint
npm run format
```

- /backend

```bash
npm run build
npm run test
npm run lint
npm run format
```

- /contract

```bash
npm run test
npm run lint
npm run format
```

- **Please check if the PR fulfills these requirements**

- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [x] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

File: app/src/components/sections/SingleTeamView/BoardOfDirectorsSection.vue
Comment (dasarathg68) [inline app/src/components/sections/SingleTeamView/BoardOfDirectorsSection.vue:1]: used useReadContract to read the owners of the expense and bank account, and the board of directors.

File: app/src/components/sections/SingleTeamView/ExpenseAccountSection.vue
Comment (dasarathg68) [inline app/src/components/sections/SingleTeamView/ExpenseAccountSection.vue:1]: used useReadContract to read the owner of expense account, board of directors and the balance of the expense account.

File: app/src/components/sections/SingleTeamView/ProposalSection.vue
Comment (dasarathg68) [inline app/src/components/sections/SingleTeamView/ProposalSection.vue:1]: useReadContract for board of directors

File: app/src/components/sections/SingleTeamView/modals/ApproveRevokeAction.vue
Comment (dasarathg68) [inline app/src/components/sections/SingleTeamView/modals/ApproveRevokeAction.vue:1]: useReadContract for isApproved & approvalCount

File: app/src/composables/bank.ts
Comment (dasarathg68) [inline app/src/composables/bank.ts:1]: removed associated composables

**Review by dasarathg68 — COMMENTED** [review]
This PR demoes the use of Wagmi hooks to useReadContract, useWriteContract, useWaitForTransactionReceipt useBalance. 
We use a combination of useWriteContract and useWaitForTransactionReceipt to provide better UX.
**Right now, all functions in Bank and Proposals have been refactored to Wagmi**

File: app/src/components/sections/SingleTeamView/ProposalCard.vue
Comment (dasarathg68) [inline app/src/components/sections/SingleTeamView/ProposalCard.vue:1]: The conclude, voteElection, voteDirective calls have been refactored to wagmi using useWriteContract+useWaitForTransactionReceipt

File: app/src/components/sections/SingleTeamView/VotingManagement.vue
Comment (dasarathg68) [inline app/src/components/sections/SingleTeamView/VotingManagement.vue:1]: Made use of useWriteContract+useWaitForTransactionReceipt for pause, unpause, transferownership

File: app/src/components/sections/SingleTeamView/BankManagement.vue
Comment (dasarathg68) [inline app/src/components/sections/SingleTeamView/BankManagement.vue:1]: used useWriteContract+useWaitForTransactionReceipt for pause, unpause and transfer ownership

**Review by hermannleboss — COMMENTED** [review]
LGTM 

I think the acceptance criteria should be
- Remove All Contracts Services
- Remove All Contracts Composable

**Pull request merged**